package.json

```diff
@@ -23,9 +23,11 @@
     }
   },
   "dependencies": {
+    "@types/react-hammerjs": "^1.0.4",
     "compression": "^1.7.4",
     "express": "^4.18.2",
     "http-proxy-middleware": "^2.0.6",
+    "react-hammerjs": "^1.0.1",
```

Comment:
lock file needs to be committed as well, I am getting an error on docker build on this branch

```
 => ERROR [runner 2/4] RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --prod --frozen-lockfile --ignore-scripts                                                                                                        1.1s
 => CANCELED [builder 3/5] RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile --ignore-scripts                                                                                                           1.1s
------                                                                                                                                                                                                                                      
 > [runner 2/4] RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --prod --frozen-lockfile --ignore-scripts:                                                                                                                   
#0 1.044  ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with package.json                                                                                                      
#0 1.044                                                                                                                                                                                                                                    
#0 1.044 Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"                                                                              
#0 1.044                                                                                                                                                                                                                                    
#0 1.044     Failure reason:
#0 1.044     specifiers in the lockfile ({"compression":"^1.7.4","express":"^4.18.2","http-proxy-middleware":"^2.0.6","vite-express":"^0.9.2","@capacitor-community/app-icon":"npm:voyager-capacitor-app-icon@^4.1.0","@capacitor/android":"5.2.3","@capacitor/app":"^5.0.6","@capacitor/assets":"^2.0.4","@capacitor/browser":"^5.0.6","@capacitor/cli":"5.2.3","@capacitor/core":"5.2.3","@capacitor/haptics":"npm:voyager-capacitor-haptics@^5.0.7","@capacitor/ios":"5.2.3","@capacitor/keyboard":"^5.0.6","@capacitor/share":"^5.0.6","@capacitor/status-bar":"^5.0.6","@emotion/babel-plugin":"^11.11.0","@emotion/css":"^11.11.2","@emotion/eslint-plugin":"^11.11.0","@emotion/react":"^11.11.1","@emotion/styled":"^11.11.0","@github/markdown-toolbar-element":"^2.2.0","@ionic/core":"npm:voyager-ionic-core@^7.2.3-tweak.1","@ionic/react":"^7.2.3","@ionic/react-router":"^7.2.3","@reduxjs/toolkit":"^1.9.5","@testing-library/jest-dom":"^6.0.0","@testing-library/react":"^14.0.0","@testing-library/user-event":"^14.4.3","@trapezedev/configure":"^7.0.10","@types/history":"^4.0.0","@types/jest":"^29.5.3","@types/js-cookie":"^3.0.3","@types/lodash":"^4.14.197","@types/mdast":"^4.0.0","@types/react":"^18.2.20","@types/react-dom":"^18.2.7","@types/react-router":"^5.1.20","@types/react-router-dom":"^5.3.3","@types/ua-parser-js":"^0.7.36","@types/uuid":"^9.0.2","@typescript-eslint/eslint-plugin":"^6.3.0","@typescript-eslint/parser":"^6.3.0","@vitejs/plugin-legacy":"^4.1.1","@vitejs/plugin-react":"^4.0.4","capacitor-android-nav-mode":"^0.0.1","capacitor-application-context":"^0.0.1","capacitor-plugin-safe-area":"^2.0.5","capacitor-set-version":"^2.0.13","capacitor-stash-media":"^0.0.6","date-fns":"^2.30.0","dexie":"^3.2.4","dexie-react-hooks":"^1.1.6","eslint":"^8.47.0","eslint-config-prettier":"^9.0.0","eslint-plugin-prettier":"^5.0.0","eslint-plugin-react":"^7.33.1","eslint-plugin-react-hooks":"^4.6.0","history":"^4.9.0","ionicons":"^7.1.2","js-cookie":"^3.0.5","jsdom":"^22.1.0","lemmy-js-client":"0.18.0","lodash":"^4.17.21","mdast-util-gfm-autolink-literal-lemmy":"^1.0.0","mdast-util-gfm-strikethrough":"^1.0.0","mdast-util-gfm-table":"^1.0.0","micromark":"^3.0.0","micromark-extension-gfm-strikethrough":"^1.0.0","micromark-extension-gfm-table":"^1.0.0","micromark-util-combine-extensions":"^1.0.0","photoswipe":"^5.3.8","prettier":"^3.0.1","pwa-asset-generator":"^6.3.1","react":"^18.2.0","react-animate-height":"^3.2.2","react-dom":"^18.2.0","react-markdown":"^8.0.7","react-redux":"^8.1.2","react-router":"^5.3.4","react-router-dom":"^5.3.4","react-textarea-autosize":"^8.5.2","react-transition-state":"^2.1.1","react-virtuoso":"^4.5.0","release-it":"^16.1.5","remark-gfm":"^3.0.1","terser":"^5.19.2","typescript":"^5.1.6","ua-parser-js":"^1.0.35","unified":"^10.1.2","unist-util-visit":"^5.0.0","use-long-press":"^3.1.5","usehooks-ts":"^2.9.1","uuid":"^9.0.0","vite":"^4.4.9","vite-plugin-pwa":"^0.16.4","vite-plugin-svgr":"^3.2.0","vitest":"^0.34.1","workbox-window":"^7.0.0"}) don't match specs in package.json ({"@capacitor-community/app-icon":"npm:voyager-capacitor-app-icon@^4.1.0","@capacitor/android":"5.2.3","@capacitor/app":"^5.0.6","@capacitor/assets":"^2.0.4","@capacitor/browser":"^5.0.6","@capacitor/cli":"5.2.3","@capacitor/core":"5.2.3","@capacitor/haptics":"npm:voyager-capacitor-haptics@^5.0.7","@capacitor/ios":"5.2.3","@capacitor/keyboard":"^5.0.6","@capacitor/share":"^5.0.6","@capacitor/status-bar":"^5.0.6","@emotion/babel-plugin":"^11.11.0","@emotion/css":"^11.11.2","@emotion/eslint-plugin":"^11.11.0","@emotion/react":"^11.11.1","@emotion/styled":"^11.11.0","@github/markdown-toolbar-element":"^2.2.0","@ionic/core":"npm:voyager-ionic-core@^7.2.3-tweak.1","@ionic/react":"^7.2.3","@ionic/react-router":"^7.2.3","@reduxjs/toolkit":"^1.9.5","@testing-library/jest-dom":"^6.0.0","@testing-library/react":"^14.0.0","@testing-library/user-event":"^14.4.3","@trapezedev/configure":"^7.0.10","@types/history":"^4.0.0","@types/jest":"^29.5.3","@types/js-cookie":"^3.0.3","@types/lodash":"^4.14.197","@types/mdast":"^4.0.0","@types/react":"^18.2.20","@types/react-dom":"^18.2.7","@types/react-router":"^5.1.20","@types/react-router-dom":"^5.3.3","@types/ua-parser-js":"^0.7.36","@types/uuid":"^9.0.2","@typescript-eslint/eslint-plugin":"^6.3.0","@typescript-eslint/parser":"^6.3.0","@vitejs/plugin-legacy":"^4.1.1","@vitejs/plugin-react":"^4.0.4","capacitor-android-nav-mode":"^0.0.1","capacitor-application-context":"^0.0.1","capacitor-plugin-safe-area":"^2.0.5","capacitor-set-version":"^2.0.13","capacitor-stash-media":"^0.0.6","date-fns":"^2.30.0","dexie":"^3.2.4","dexie-react-hooks":"^1.1.6","eslint":"^8.47.0","eslint-config-prettier":"^9.0.0","eslint-plugin-prettier":"^5.0.0","eslint-plugin-react":"^7.33.1","eslint-plugin-react-hooks":"^4.6.0","history":"^4.9.0","ionicons":"^7.1.2","js-cookie":"^3.0.5","jsdom":"^22.1.0","lemmy-js-client":"0.18.0","lodash":"^4.17.21","mdast-util-gfm-autolink-literal-lemmy":"^1.0.0","mdast-util-gfm-strikethrough":"^1.0.0","mdast-util-gfm-table":"^1.0.0","micromark":"^3.0.0","micromark-extension-gfm-strikethrough":"^1.0.0","micromark-extension-gfm-table":"^1.0.0","micromark-util-combine-extensions":"^1.0.0","photoswipe":"^5.3.8","prettier":"^3.0.1","pwa-asset-generator":"^6.3.1","react":"^18.2.0","react-animate-height":"^3.2.2","react-dom":"^18.2.0","react-markdown":"^8.0.7","react-redux":"^8.1.2","react-router":"^5.3.4","react-router-dom":"^5.3.4","react-textarea-autosize":"^8.5.2","react-transition-state":"^2.1.1","react-virtuoso":"^4.5.0","release-it":"^16.1.5","remark-gfm":"^3.0.1","terser":"^5.19.2","typescript":"^5.1.6","ua-parser-js":"^1.0.35","unified":"^10.1.2","unist-util-visit":"^5.0.0","use-long-press":"^3.1.5","usehooks-ts":"^2.9.1","uuid":"^9.0.0","vite":"^4.4.9","vite-plugin-pwa":"^0.16.4","vite-plugin-svgr":"^3.2.0","vitest":"^0.34.1","workbox-window":"^7.0.0","@types/react-hammerjs":"^1.0.4","compression":"^1.7.4","express":"^4.18.2","http-proxy-middleware":"^2.0.6","react-hammerjs":"^1.0.1","vite-express":"^0.9.2"})
------
Error: failed to fetch an image or build from source: error building: failed to solve: executor failed running [/bin/sh -c pnpm install --prod --frozen-lockfile --ignore-scripts]: exit code: 1
```

Reply:
ah alright. I had used yarn to install these. i'll install again using `pnpm`

Reply:
also, I think react-hammerjs is not needed here. there's some components that use `useLongPress`.

See `src/features/labels/links/CommunityLink.tsx` for example. Maybe that can be used here for the same functionality